src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1231,7 +1231,11 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
 			si_udp_logdbg("stopping de-muxing packets to port %d", ntohs(port));
 			m_port_map_lock.lock();
 			m_port_map.erase(std::remove(m_port_map.begin(), m_port_map.end(), port), m_port_map.end());
-			m_sockopt_mapped = false;
+			if (m_port_map.size()) {
+				m_sockopt_mapped = true;
+			} else {
+				m_sockopt_mapped = false;
+			}
```

Comment:
Hi,
There is not reason to change _m_sockopt_mapped_ to true while removing an item from the map.
Additionally, we should call _set_rx_packet_processor()_ only when we erase the last element from the map (similar approach should also be applied to UDP_MAP_ADD).
Please see patch suggestion at redmine.

Reply:
Agree with Liran's comment, Sergey, please check suggested patch.